universe/NamedValueRefManager.cpp

```diff
@@ -156,16 +156,17 @@ namespace {
                               std::string&& valueref_name, std::unique_ptr<VR>&& vref)
     {
         TraceLogger() << "Register " << label << " valueref for " << valueref_name << ": " << vref->Description();
-        if (container.count(valueref_name)>0) {
+        if (container.count(valueref_name) > 0) {
             TraceLogger() << "Skip registration for already registered " << label << " valueref for " << valueref_name;
             TraceLogger() << "Number of registered " << label << " ValueRefs: " << container.size();
             return;
         }
         TraceLogger() << "RegisterValueRefImpl Check invariances for info. Then add the value ref in a thread safe way.";
-        const std::lock_guard<std::mutex> lock(mutex);
+        std::scoped_lock lock(mutex);
```

Comment:
Use of scoped lock is not strictly necessary - no other lock call on that mutex object can happen. Single lock operation on this mutex and the function does not call anything which registers value refs itself ("getters" and modification of the singleton's containers ).

Reply:
If you want to remove some locks, that would be best discussed in a separate pull request

Reply:
Was mistaking scoped lock for recursive lock. I think a lock here is strictly necessary. Case closed.